src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,97 @@
+})
+export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() public segment: Segment
+  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+
+  private readonly defaultAssetForThumbnail: string = 'assets/sofie-logo.svg'
+  protected mediaDuration: number = 0
+  private configurationMediaPreviewUrl: string
+  private configurationServiceSubscription: Subscription
+  private executeActionSubscription: Subscription
+  private serverPostrollDuration: number = 0
+
+  constructor(
+    private readonly actionService: ActionService,
+    private readonly configurationService: ConfigurationService,
+    private readonly mediaStateService: MediaStateService
+  ) {}
+
+  public ngOnInit(): void {
+    this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((configuration: StudioConfiguration) => {
+      this.configurationMediaPreviewUrl = configuration.data.settings.mediaPreviewUrl
+      this.serverPostrollDuration = configuration.data.blueprintConfiguration.ServerPostrollDuration
+    })
+    void this.updateMediaDataDuration()
+  }
+
+  private async updateMediaDataDuration(): Promise<void> {
+    if (this.segment.metadata?.miniShelfVideoClipFile !== undefined) {
+      let media: Media | undefined = await this.mediaStateService.getMedia(this.segment.metadata?.miniShelfVideoClipFile)
+      if (!media || !media.duration) {
+        return
+      }
+      if (media.duration < this.serverPostrollDuration) {
+        this.mediaDuration = this.NaN
+      } else {
+        this.mediaDuration = media.duration - this.serverPostrollDuration
+      }
+    }
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if ('segment' in changes) {
+      void this.updateMediaDataDuration()
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.configurationServiceSubscription?.unsubscribe()
+    this.executeActionSubscription?.unsubscribe()
+  }
+
+  protected get mediaPreviewUrl(): string {
+    const url: string = `${this.configurationMediaPreviewUrl}/media/thumbnail/${this.segment.metadata?.miniShelfVideoClipFile}`
+    return this.configurationMediaPreviewUrl ? url : this.defaultAssetForThumbnail
+  }
+
+  public getSanitizedTitle(): string {
+    return this.segment.name
+      .replace(/<[^>]*>/g, '') // remove html tags
+      .replace(/(\r\n|\n|\r)/gm, ' ') // remove newlines
+      .replace(/_/g, ' ') // replace underscores with spaces;
+      .replace(/&nbsp;/g, ' ') // remove &nbsp;
+      .replace(/\s+/g, ' ') // remove extra spaces
+      .replace(/&amp;/g, '&') // remove &amp;
+      .replace(/&quot;/g, '"') // remove &quot;
+      .replace(/&apos;/g, "'") // remove &apos;
+      .trim() // and finally trim;
+  }
+
+  protected executeAction(): void {
+    if (!this.videoClipAction) {
+      return
+    }
+    this.executeActionSubscription = this.actionService.executeAction(this.videoClipAction.id, this.segment.rundownId).subscribe()
```

Comment:
You don't need to store the subscribtion for this.
The angular framework unsubscribes automatically from this :) 

So this can be written as:
```
public executeAction(): void {
  this.actionService.executeAction(this.videoClipAction.id, this.segment.rundownId).subscribe()
}
```

Also, why is it protected? You seems to be using `procteted` and `public` arbitrarily on the methods in here?